### PR TITLE
heminetwork update

### PR DESCRIPTION
**Pull request by ClaytonNorthey92**
update heminetwork/tbc dependency

updated function signatures in calls

ran localnet tests, they pass ✅ 

should be merged with https://github.com/hemilabs/optimism/pull/4

### CODE REVIEW

File: cmd/geth/main.go [hunk 1]
```diff
@@ -405,70 +405,62 @@ func startNode(ctx *cli.Context, stack *node.Node, backend ethapi.Backend, isCon
...
 
-		log.Info("Finished initial indexing", "initHeight", initHeight)
-	}
+		if err := vm.TBCIndexer.SyncIndexersToHash(ctx.Context, &bestHash); err != nil {
+			log.Crit(fmt.Sprintf("could not SyncIndexersToHash: %v", err))
+		}
 
-	si := vm.TBCIndexer.Synced(ctx.Context)
+		if err := vm.TBCIndexer.TxIndexer(ctx.Context, &bestHash); err != nil {
+			log.Crit(fmt.Sprintf("coult not TxIndexer: %v", err))
```
Comment (joshuasing) [inline cmd/geth/main.go]: I think this has a typo:
```suggestion
			log.Crit(fmt.Sprintf("could not TxIndexer: %v", err))
```

File: cmd/geth/main.go [hunk 2]
```diff
@@ -405,70 +405,62 @@ func startNode(ctx *cli.Context, stack *node.Node, backend ethapi.Backend, isCon
...
+		}
 
-	if si.UtxoHeight < initHeight {
-		log.Crit("TBC did not index UTXOs to initHeight!",
-			"utxoIndexHeight", si.UtxoHeight, "initHeight", initHeight)
-	}
+		if err := vm.TBCIndexer.UtxoIndexer(ctx.Context, &bestHash); err != nil {
+			log.Crit(fmt.Sprintf("coult not UTXOIndexer: %v", err))
```
Comment (joshuasing) [inline cmd/geth/main.go]: I think this has a typo:
```suggestion
			log.Crit(fmt.Sprintf("could not UTXOIndexer: %v", err))
```

File: core/vm/contracts.go [hunk 3]
```diff
@@ -129,95 +125,66 @@ func ProgressTip(ctx context.Context, currentTimestamp uint32) {
...
 		}
 	}
 }
 
-func TBCIndexTxs(ctx context.Context, tipHeight uint64) error {
-	var h uint64
-	firstSync := false
-	// Get current indexed height
-	he, err := TBCIndexer.DB().MetadataGet(ctx, tbc.TxIndexHeightKey)
+func TBCIndexTxs(ctx context.Context) error {
```
Comment (joshuasing) [inline core/vm/contracts.go]: Would it be useful to wrap the errors in this function to add more detail for easier debugging?

File: core/vm/contracts.go [hunk 4]
```diff
@@ -129,95 +125,66 @@ func ProgressTip(ctx context.Context, currentTimestamp uint32) {
...
+
 	return nil
 }
 
-func TBCIndexUTXOs(ctx context.Context, tipHeight uint64) error {
-	var h uint64
-	firstSync := false
-	// Get current indexed height
-	he, err := TBCIndexer.DB().MetadataGet(ctx, tbc.UtxoIndexHeightKey)
+func TBCIndexUTXOs(ctx context.Context) error {
```
Comment (joshuasing) [inline core/vm/contracts.go]: Would it be useful to wrap the errors in this function to add more detail for easier debugging?

File: core/vm/contracts.go (hunk 3 above)
Comment (ClaytonNorthey92) [reply core/vm/contracts.go]: @joshuasing I think we should delete this function along with `TBCIndexUTXOs`; I was talking to @max-sanchez last week, these are unused and soon-to-be deleted.  probably not worth the fuss.   I will remove 👍

**Review by joshuasing — APPROVED** [review]

**Pull request merged**